### PR TITLE
Install package.json if present

### PR DESCRIPTION
**Pull request by evanstachowiak**
A lot of nodejs projects have a package.json. Add support to install a package.json if the path is set as a variable.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -19,3 +19,6 @@ nodejs_npm_global_packages: []
 #    version: 0.9.3
 #  # Install the latest stable release of a package.
 #  - name: node-sass
+
+# The path of a package.json
+nodejs_package_json_path:
```
Comment (geerlingguy) [inline defaults/main.yml]: In this definition, can you set it to an empty string so it's more obvious a string should be entered?

```yaml
nodejs_package_json_path: ""
```

Also, can you add the variable, then a sentence or two describing what it does, to the README.md?

File: tasks/main.yml [hunk 2]
```diff
@@ -48,3 +48,8 @@
     NODE_PATH: "{{ npm_config_prefix }}/lib/node_modules"
     NPM_CONFIG_UNSAFE_PERM: "{{ npm_config_unsafe_perm }}"
   with_items: "{{ nodejs_npm_global_packages }}"
+
+- name: Install package.json
+  npm:
+    path: "{{ package_json_path }}"
+  when: nodejs_package_json_path is defined
```
Comment (geerlingguy) [inline tasks/main.yml]: If setting the var to an empty string, this when clause would need to be come:

```yaml
when: nodejs_package_json_path is defined and nodejs_package_json_path
```

**Comment by geerlingguy** [comment]
As a side benefit—defining the default variable will also let the tests pass.

**Comment by evanstachowiak** [comment]
hey @geerlingguy, i've addressed your changes, let me know what you think.

File: README.md [hunk 3]
```diff
@@ -39,6 +39,10 @@ Add a list of npm packages with a `name` and (optional) `version` to be installe
       # Install the latest stable release of a package.
       - name: node-sass
 
+Set a path pointing to a package.json. This will install all of the packages globally.
+
+    nodejs_npm_global_packages: "/var/www/application/package.json"
```
Comment (geerlingguy) [inline README.md]: This variable needs to be: `nodejs_package_json_path: ""`

Also, please place the description of the variable below the variable itself.

**Comment by evanstachowiak** [comment]
oops, my bad with the variable name, fixed now.

File: README.md (hunk 3 above)
Comment (evanstachowiak) [inline README.md]: oops, my bad with the variable name, fixed now.

**Pull request merged**